Automattic/wp-calypso
# Improve messages to user when deleting a P2 workspace

#### Changes proposed in this Pull Request
* On P2 workspaces, this PR adds a warning for the user that a workspace with P2s in it cannot be deleted and that they need to go delete the P2(s) in the workspace first.
* It adds a react-query `useP2HubP2sQuery` that can get the P2s a workspace contains. It can likely be useful for other things related to workspaces in the future.
* It adds a HOC wrapping the `useP2HubP2sQuery`. It's only purpose is to hold the number of P2s in a workspace. 

#### Testing instructions
You'll need a P2 workspace and an ordinary wpcom site to test with.
* Go to `http://calypso.localhost:3000/settings/general/<yourp2workspace>.wordpress.com` and click "Delete your site permanently". You should get a dialog with the number of P2s in your workspace and a warning that they need to be deleted before you can delete the workspace.
* On an ordinary site, do the same as above. You should get no dialog if the site is not on a paid plan and if it is, then you should get a warning that the site is on the plan and that you should cancel the upgrades before deleting.


With only 1 P2 in the workspace:
<img width="642" alt="Screen Shot 2022-02-01 at 14 10 58" src="https://user-images.githubusercontent.com/193283/152044296-fd8c3ec8-d59e-49c2-9085-86231c57e333.png">

With 13 P2s in the workspace:
<img width="710" alt="Screen Shot 2022-02-01 at 14 10 39" src="https://user-images.githubusercontent.com/193283/152044338-937b45f3-bc97-4f88-92e1-ef09e479961b.png">



See 2840-gh-Automattic/p2
